core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -1207,19 +1240,36 @@ impl BundleStorage {
             .map(|b| b.len())
             .sum::<usize>();
 
-        let to_insert = deserialized_bundles
-            .into_iter()
-            .take(bundles_to_insert_count);
-        if push_back {
-            deque.extend(to_insert)
-        } else {
-            to_insert.for_each(|b| deque.push_front(b));
+        match storage {
+            Either::Left(unprocessed_storage) => {
+                deserialized_bundles
```

Comment:
instead of for_each + push, lets use `extend`. Also, let's just calculate the priority inline here instead of calling out to a function. While a bit more verbose this will make it easier to audit as the priority calculation code is not used anywhere else so we should only consider it in context.